packages/core-api/src/routing/RouteRef.ts

```diff
@@ -58,42 +78,58 @@ export function createRouteRef<
   params?: ParamKey[];
   /** @deprecated Route refs no longer decide their own path */
   path?: string;
+  /** @deprecated Route refs no longer decide their own icon */
   icon?: IconComponent;
+  /** @deprecated Route refs no longer decide their own title */
   title: string;
-}): RouteRef<Params> {
-  return new AbsoluteRouteRef<Params>(config);
+}): RouteRef<OptionalParams<Params>> {
+  return new RouteRefImpl<OptionalParams<Params>>({
+    ...config,
+    params: (config.params ?? []) as ParamKeys<OptionalParams<Params>>,
+  });
 }
 
-export class ExternalRouteRef<Optional extends boolean = true> {
-  readonly optional: boolean;
+export class ExternalRouteRefImpl<
+  Params extends AnyParams,
+  Optional extends boolean
+> extends RouteRefBaseBase {
+  readonly [routeRefType] = 'external';
```

Comment:
Can this be assigned by the superclass, to avoid duplication?

Reply:
Not without more type shenanigans as these need to implement the specific type. I added in `implements` for clarity though